hyperspy/tests/component/test_component_active_array.py

```diff
@@ -52,15 +53,14 @@ def test_toggle_array(self):
         g._create_arrays()
         g._active_array[2, 0] = False
         g._active_array[0, 0] = False
-        g._toggle_connect_active_array(False)
-        nt.assert_false(g._active_is_multidimensional)
-        nt.assert_true(
-            np.all(
-                g.A.as_signal('values').data == np.zeros(
-                    (3, 3))))
-        nt.assert_true(np.all(
-            g._active_array == np.array([[0, 1, 1], [1, 1, 1], [0, 1, 1]], dtype=bool)))
-        g._toggle_connect_active_array(True)
+        with stash_active_state([g]):
+            g.active_is_multidimensional = False
+            nt.assert_false(g._active_is_multidimensional)
+            nt.assert_true(
+                np.all(
+                    g.A.as_signal('values').data == np.zeros(
+                        (3, 3))))
+            nt.assert_equal(g._active_array, None)
```

Comment:
I would interpret the code as checking that `as_signal` correctly puts nans where the active array is `False`. In fact it only checks those particular "spots", so even if the result array was filled with nans, it would still pass. Also, there used to be a test that explicitly checked the `_active_array`. So I'd like to keep it there - it may have been put there for a specific corner-case, and I just don't remember...